motor/core.py

```diff
@@ -2034,3 +2041,28 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     async def get_keys(self):
         cursor_class = create_class_with_framework(AgnosticCursor, self._framework, self.__module__)
         return cursor_class(self.delegate.get_keys(), self)
+
+    def _lazy_init(self, kwargs):
+        if not self.delegate:
+            self.delegate = self.delegate.watch(
+                **unwrap_kwargs_session(kwargs, endswith="MotorClientEncryption")
+            )
+
+    def _create_encrypted_collection(self, *args, **kwargs):
+        # This method is run on a thread.
+        self._lazy_init(kwargs)
+        return self.delegate.create_encrypted_collection(*args, **kwargs)
+
+    async def create_encrypted_collection(self, *args, **kwargs):
+        collection_class = create_class_with_framework(
+            AgnosticCollection, self._framework, self.__module__
+        )
+        loop = self.get_io_loop()
+        coll, ef = await self._framework.run_on_executor(
+            loop, self._create_encrypted_collection, *args, **kwargs
+        )
+        try:
+            coll = await coll
```

Comment:
You can't await a Collection object so what are we trying to do here?

Reply:
Removed.